pennylane/ops/op_math/condition.py

```diff
@@ -430,3 +446,128 @@ def wrapper(*args, **kwargs):
         )
 
     return wrapper
+
+
+@functools.lru_cache
+def _get_cond_qfunc_prim():
+    """Get the cond primitive for quantum functions."""
+
+    # JAX should be installed if capture is enabled
+    import jax  # pylint: disable=import-outside-toplevel
+
+    cond_prim = jax.core.Primitive("cond")
+    cond_prim.multiple_results = True
+
+    @cond_prim.def_impl
+    def _(condition, elifs_conditions, *args, jaxpr_true, jaxpr_false, jaxpr_elifs):
+
+        def run_jaxpr(jaxpr, *args):
+            return jax.core.eval_jaxpr(jaxpr.jaxpr, jaxpr.consts, *args)
+
+        def true_branch(args):
+            return run_jaxpr(jaxpr_true, *args)
+
+        def elif_branch(args, elifs_conditions, jaxpr_elifs):
+            if not jaxpr_elifs:
+                return false_branch(args)
+            pred = elifs_conditions[0]
+            rest_preds = elifs_conditions[1:]
+            jaxpr_elif = jaxpr_elifs[0]
+            rest_jaxpr_elifs = jaxpr_elifs[1:]
+            if pred:
+                return run_jaxpr(jaxpr_elif, *args)
+            return elif_branch(args, rest_preds, rest_jaxpr_elifs)
+
+        def false_branch(args):
+            if jaxpr_false is not None:
+                return run_jaxpr(jaxpr_false, *args)
+            return ()
+
+        if condition:
+            return true_branch(args)
+        if elifs_conditions.size > 0:
+            return elif_branch(args, elifs_conditions, jaxpr_elifs)
+        return false_branch(args)

```

Comment:
More from a code design point, but the fact `elif_branch` can also call the `false_branch` is a bit unexpected.

Reply:
It's not so strange to me because each `elif` condition is logically inserted between `true_fn` and `false_fn`, so if there are no `elifs` conditions we simply move to the `false_branch`. However, I agree that maybe it's better to move this logic outside of a function called `elif_branch`